CodeConverter.Web/wwwroot/js/index.js

```diff
@@ -32,20 +32,19 @@ var app = new Vue({
             this.errorMessageOnResponse = "";
             this.showErrors = false;
 
-            var xhttp = new XMLHttpRequest();
             var capturedObj = this;
+            axios.post('/api/converter/', {
+                code: this.inputCode,
+                requestedConversion: this.requestedConversion
+            })
+            .then(function (response) {
+                capturedObj.converterCallInflight = false;
+                if (200 === response.status) {
+                    capturedObj.convertedCode = response.data.convertedCode;
 
-            xhttp.onreadystatechange = function () {
-                if (this.readyState === 4) {
```

Comment:
If the aim was to make this simpler to read, it might have been a bit more lightweight to use the onLoad method which only fires in readystate 4.
Or you could just extract the 4 as a nicely named constant, or set it to `XMLHttpRequest.DONE ||  XMLHttpRequest.READYSTATE_COMPLETE` (the latter part if for IE10 compatibility).

Given the simplicity of the site, I think we're unnecessarily introducing more possible failure states by adding dependencies, especially in terms of security vulnerabilities. That said, it looks like axios is the best choice of the ones available.